sklearn/utils/estimator_checks.py

```diff
@@ -2858,17 +2934,20 @@ def check_outliers_fit_predict(name, estimator_orig):
             assert_raises(ValueError, estimator.fit_predict, X)
 
 
-def check_fit_non_negative(name, estimator_orig):
+def check_fit_non_negative(name, estimator_orig, strict_mode=True):
```

Comment:
This is the only "partially strict check" that I implemented for now. I want to make the PR about the design rather than about the specific details of each checks.